### PR TITLE
Refactor MessageArgs

### PR DESCRIPTION
**Pull request by jeremiah-corrado**
This PR makes a few changes to improve the ergonomics of argument parsing:

* added `this(key)` accessor to `MessageArgs`, allowing things like `const x = msgArgs["x"]`
* modified `getGenericTypedArrayEntry` to accept `ParamObj` directly, allowing things like: `getGenericTypedArrayEntry(msgArgs["name"], st)`
* added some more generic methods for parsing scalars, and json lists
  * a json list can be easily parsed as a tuple (if size is known at compile time), a Chapel array, or a Chapel list
* removed unused/uneeded methods from `MessageArgs`
* removed `objType` field from `MessageArgs`as it was only used in Efunc2 and was easily confused with the ObjType defined in serverConfig (resolves https://github.com/Bears-R-Us/arkouda/issues/2522)

Most of the old MessageArgs API still exists, so message handlers can be converted to use the new API over time. As an example, ManipulationMsg was converted to use the new API.

Also cleaned up IO compat modules (most of those changes were moved to #3365), leaving only symbols related to the `ioendian`->`endianness` change.

### CODE REVIEW

File: src/ArgSortMsg.chpl [hunk 1]
```diff
@@ -338,7 +338,7 @@ module ArgSortMsg
         }
       }
       try! asLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),
-                             "argsort time = %i".doFormat(Time.timeSinceEpoch().totalSeconds() - t1));
+                             "argsort time = %i".format(Time.timeSinceEpoch().totalSeconds() - t1));
```
Comment (bmcdonald3) [inline src/ArgSortMsg.chpl:341]: I wonder if it makes sense to pull this `doFormat` -> `format` change into it's own PR given that it's a huge number of changes dirtying this PR and it's only a single commit, so could easily be picked out. What do you think?

Comment (jeremiah-corrado) [reply src/ArgSortMsg.chpl:341]: Yeah, that's a good point. I can make a separate PR for it.

Comment (jeremiah-corrado) [reply src/ArgSortMsg.chpl:341]: PR here: https://github.com/Bears-R-Us/arkouda/pull/3365

**Comment by jeremiah-corrado** [comment]
CI failed to install dependencies. Could someone please restart when you get a chance.

**Review by ajpotts — APPROVED** [review]
Thanks for doing all this!

**Review by jaketrookman — APPROVED** [review]
Looks good

**Review by bmcdonald3 — APPROVED** [review]
Love it <3

File: src/Message.chpl [hunk 2]
```diff
@@ -188,31 +188,129 @@ module Message {
...
+                        throw err();
+                    }
+                }
+            } else {
+
+                try {
+                    return parseJson(this.val, size, t);
+                } catch {
+                    throw err();
+                }
```
[the remark this answers is not in the document]
Comment (jeremiah-corrado) [reply src/Message.chpl:230]: no, I think I had some comments or something on those lines that got removed. I can fix that real quick.

**Review by bmcdonald3 — APPROVED** [review]

**Review by stress-tess — APPROVED** [review]
looks good!! Thanks jeremiah!

**Comment by jeremiah-corrado** [comment]
Thanks for the reveiws everyone! It looks like CI failed again due to dependency installation timeouts. Could someone please re-add to the merge queue :)

**Pull request merged**